### PR TITLE
Delete `GT_FIELD`

### PR DESCRIPTION
**Pull request by SingleAccretion**
With the removal `ASG`, `FIELD` is the only location node that does not have a `STORE` equivalent, thus this change removes it in favor of importing `IND/BLK(FIELD_ADDR)`.

Diffs:
1) Code - a number of quirks had to be applied in order to keep those down, but many still remain. Common theme is lack of forward substitution due to the 16 node limit. There are a number of other reasons as well, e. g. on x86 we now spill a `FIELD_ADDR` return buffer argument for a GDV call arg because it is correctly marked with `GTF_EXCEPT`.
2) TP - a bit of a significant regression due to the IR size increase, payed for with #85544.

A number of `TODO-FIELD`s were added, I intend to address them separately.

### CODE REVIEW

File: src/coreclr/jit/simd.cpp [hunk 1]
```diff
@@ -472,73 +472,31 @@ void Compiler::setLclRelatedToSIMDIntrinsic(GenTree* tree)
...
+    var_types         op2Type      = op2->TypeGet();
+    GenTreeFieldAddr* op1Addr      = op1->Addr()->AsFieldAddr();
+    GenTreeFieldAddr* op2Addr      = op2->Addr()->AsFieldAddr();
+    unsigned          op1EndOffset = op1Addr->gtFldOffset + genTypeSize(op1Type);
+    unsigned          op2Offset    = op2Addr->gtFldOffset;
+    if ((op1Type == op2Type) && (op1EndOffset == op2Offset) && op1Addr->IsInstance() && op2Addr->IsInstance() &&
+        GenTree::Compare(op1Addr->GetFldObj(), op2Addr->GetFldObj()))
     {
         return true;
     }
```
Comment (jakobbotsch) [inline src/coreclr/jit/simd.cpp:502]: What if `op1Addr` and `op2Addr` are calls to the same function?

Comment (SingleAccretion) [reply src/coreclr/jit/simd.cpp:502]: Good point. This is too permissive, I will revert it back to the original logic.

**Review by jakobbotsch — APPROVED** [review]
LGTM. Nice to see these gone.

**Comment by jakobbotsch** [comment]
Failure is known.

**Pull request merged**